### PR TITLE
[runtime env] [1.11.0 release cherry-pick] fix bug where pip options don't work in `requirements.txt`

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
This PR cherry-picks the fix https://github.com/ray-project/ray/pull/22065 to the 1.11.0 release branch.

Merging this PR requires two approvals:  -- one committer and the TL owning the component (@edoakes).  

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by mwtian — APPROVED** [review]

**Review by edoakes — APPROVED** [review]

**Comment by architkulkarni** [comment]
@mwtian Looks like this has the necessary approvals and passed CI; I'll let you merge this whenever you're ready.

**Pull request merged**
